File: python/ajustado/src/app/modules/auth/dtos/inputs/login_auth_dto.py
from pydantic import BaseModel, EmailStr, Field, field_validator

# Common
from app.common.messages import MESSAGES

class LoginAuthDto(BaseModel):
    email: EmailStr
    senha: str = Field(..., min_length=8, json_schema_extra={"error_message": MESSAGES['validation']['min_length']('senha', 8)})

File: typescript/fastify/src/app/modules/auth/dtos/inputs/login-auth.dto.ts
import { IsString, MinLength, IsEmail } from "class-validator";

// Messages
import { MESSAGES } from "@/common/messages";

export class LoginAuthDto {
  @IsEmail({}, { message: MESSAGES.validation.IsEmail("email") })
  email: string;

  @IsString({
    message: MESSAGES.validation.IsString("senha"),
  })
  @MinLength(8, { message: MESSAGES.validation.MinLength("senha", 8) })
  senha: string;
}
